File: ResQ-Backend/src/main/java/com/livewithoutthinking/resq/dto/LoginResponse.java
package com.livewithoutthinking.resq.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class LoginResponse {

    private Integer userId;
    private String token;
    private String fullName;
    private String role;
    private boolean onShift;

}

File: resq_app/lib/models/auth/login_response.dart
class LoginResponse {
  final int userId;
  final String token;
  final String userName;
  final String role;
  final bool isOnline;

  LoginResponse({
    required this.userId,
    required this.token,
    required this.userName,
    required this.role,
    required this.isOnline,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      userId: json['userId'],
      token: json['token'],
      userName: json['fullName'],
      role: json['role'],
      isOnline: json['onShift'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'token': token,
      'fullName': userName,
      'role': role,
      'onShift': isOnline,
    };
  }
}

LoginResponse? loginResponse;
